python/adbc_driver_manager/adbc_driver_manager/_lib.pyx

```diff
@@ -498,6 +511,11 @@ cdef class ArrowArrayStreamHandle:
         """The address of the ArrowArrayStream."""
         return <uintptr_t> &self.stream
 
+    def _to_capsule(self):
+        return cpython.PyCapsule_New(
+            &self.stream, 'arrowarraystream', pycapsule_stream_deleter
```

Comment:
This will not keep `self` alive and therefore the capsule pointer may become unreachable.

I think the capsule needs to own a malloc'ed ArrowArrayStream pointer. @lidavidm Thoughts?

Reply:
> Also, once we have a standard capsule format, I'm not sure `ArrowArrayStreamHandle` is still useful.

Sure, it was a stopgap.

> I think the capsule needs to own a malloc'ed ArrowArrayStream pointer.

Either that or it should keep the Python object itself alive (which may be desirable since it'll also keep associated ADBC state alive)

Reply:
Keeping the Python object alive from the capsule would probably be harder?

Keep in mind any producer of the capsule format will need to reimplement its own destructor function.

Reply:
Isn't it a bug if the ADBC-produced capsule needs to keep a Python object alive, though? The release callback should take care of the lifetime of any required state.

Reply:
I was thinking of just stuffing it behind the stream above, yes. https://github.com/apache/arrow-adbc/pull/702#issuecomment-1561646567

Reply:
I suppose then, yes, it would be a malloc'd stream, not the stream pointer here. Sorry for getting mixed up.